app/population/middleware/set-body.free-spaces.js

```diff
@@ -0,0 +1,48 @@
+const populationService = require('../../../common/services/population')
+
+async function setBodyFreeSpaces(req, res, next) {
+  try {
+    const { populationId } = req.body.population
+
+    if (!populationId) {
+      req.body.population.freeSpaces = {
+        date: req.params.date,
+      }
+      return next()
+    }
+
+    const {
```

Comment:
Is there a need to destructure all of these variables and pass them in again?

Is `req.body.population.freeSpaces` used anywhere else other than the view? 

It might be good to jump on a call and walk through some of this as I think it strays a little from how things have been set within the middleware/controller/template layers.

Reply:
what about something like?

```js
const { camelCase } = require('lodash')
...
const population = await populationService.getById(populationId)
const freeSpaces = Object.keys(population).reduce((acc, key) => {
  acc[camelCase(key)] = pop[key]
  return acc
}, {})
```

Reply:
I think my question is more around whether we actually _need_ to do this?